src/main/java/org/folio/rest/impl/HoldingsStorageAPI.java

```diff
@@ -408,4 +431,60 @@ private boolean isUUID(String id) {
       return false;
     }
   }
+
+  private CompletableFuture<Void> updateItemEffectiveCallNumbersByHoldings(AsyncResult<SQLConnection> connection, PostgresClient postgresClient, HoldingsRecord holdingsRecord) {
+    CompletableFuture<Void> future = new CompletableFuture<>();
+    Criterion criterion = new Criterion(
+      new Criteria().addField("holdingsRecordId")
+        .setJSONB(false).setOperation("=").setVal(holdingsRecord.getId()));
+    postgresClient.get(ITEM_TABLE, Item.class, criterion, false, false, response -> {
+      updateEffectiveCallNumbers(connection, postgresClient, response.result().getResults(), holdingsRecord).thenAccept(voidResult -> {
+        future.complete(null);
+      });
+    });
+    return future;
+  }
+
+  private CompletableFuture<Void> updateEffectiveCallNumbers(AsyncResult<SQLConnection> connection, PostgresClient postgresClient, List<Item> items, HoldingsRecord holdingsRecord) {
+    CompletableFuture<Void> allItemsUpdated = new CompletableFuture<>();
+    List<Function<SQLConnection, Future<UpdateResult>>> batchFactories = items
+      .stream()
+      .map(item -> updateItemEffectiveCallNumber(item, holdingsRecord))
+      .map(item -> updateSingleBatchFactory(ITEM_TABLE, item.getId(), item, postgresClient))
+      .collect(Collectors.toList());
+
+    SQLConnection connectionResult = connection.result();
```

Comment:
Maybe don't unwrap this here so you can pass it to the apply as is and directly to the PostgresClient without having to rewrap it in a `succeededFuture` on use.